analysis/analysis.go

```diff
@@ -413,20 +415,24 @@ func (c *AnalysisController) asssessRunStatus(run *v1alpha1.AnalysisRun) v1alpha
 				everythingCompleted = false
 			} else {
 				// otherwise, remember the worst status of all completed metric results
-				if worstStatus == "" {
+				if worstStatus == "" || analysisutil.IsWorse(worstStatus, metricStatus) {
 					worstStatus = metricStatus
-				} else {
-					if analysisutil.IsWorse(worstStatus, metricStatus) {
-						worstStatus = metricStatus
+					_, message := assessMetricFailureInconclusiveOrError(metric, *result)
```

Comment:
Instead of calling `assessMetricFailureInconclusiveOrError` again to get the message, we can pass the message back through `assessMetricStatus` as another return value.

Reply:
Passing a message through assessMetricStatus caused some unexpected behavior, so it's easier to have a separate method instead.